cli/src/test/java/org/commonjava/maven/ext/cli/CliTest.java

```diff
@@ -185,83 +187,210 @@ public void checkLocalRepositoryWithDefaultsAndModifiedUserSettings() throws Exc
 
         ManipulationSession session = (ManipulationSession) FieldUtils.readField( c, "session", true );
-        MavenSession ms = (MavenSession)FieldUtils.readField( session, "mavenSession", true );
+        MavenSession ms = (MavenSession) FieldUtils.readField( session, "mavenSession", true );
 
         assertEquals( ms.getLocalRepository().getBasedir(), folder.toString() );
-        assertEquals( ms.getRequest().getLocalRepository().getBasedir(),
-                      ms.getRequest().getLocalRepositoryPath().toString() );
+        assertEquals( ms.getRequest().getLocalRepository().getBasedir(), ms.getRequest().getLocalRepositoryPath().toString() );
     }
 
-
     @Test
     public void checkUnknownProperty() throws Exception
     {
         File folder = temp.newFolder();
-        File target = temp.newFile( );
+        File target = temp.newFile();
         // Locate the PME project pom file. Use that to verify inheritance tracking.
-        Files.copy( Paths.get( INTEGRATION_TEST.toString(),  "pom.xml" ), target.toPath(), StandardCopyOption.REPLACE_EXISTING );
+        Files.copy( Paths.get( INTEGRATION_TEST.toString(), "pom.xml" ), target.toPath(), StandardCopyOption.REPLACE_EXISTING );
 
         Cli c = new Cli();
-        c.run( new String[]{"-d",
-                        "--settings=" + getClass().getResource("/settings-test.xml").getFile(),
-                        "-Dmaven.repo.local=" + folder.toString(),
-                        "-DUNKNOWN_PROPERTY=DUMMY",
-                        "-Prun-its",
-                        "--file",
+        c.run( new String[] { "-d", "--settings=" + getClass().getResource( "/settings-test.xml" ).getFile(),
+                        "-Dmaven.repo.local=" + folder.toString(), "-DUNKNOWN_PROPERTY=DUMMY", "-Prun-its", "--file",
                         target.getCanonicalPath() } );
 
-        assertTrue ( systemOutRule.getLog().contains( "Unknown configuration value UNKNOWN_PROPERTY" ));
+        assertTrue( systemOutRule.getLog().contains( "Unknown configuration value UNKNOWN_PROPERTY" ) );
+    }
+
+    @Test
+    public void checkVersionAndExit()
+    {
+        expectedSystemExit.expectSystemExitWithStatus( 0 );
+        Cli.main( new String[] { "--version" } );
+        assertTrue( systemOutRule.getLog().contains( "PME CLI" ) );
+        assertTrue( systemOutRule.getLog().contains( "SHA" ) );
+    }
+
+    @Test
+    public void checkDependencies()
+    {
+        final File root = Paths.get( ROOT_DIRECTORY.toString(), "pom.xml" ).toFile();
+
+        expectedSystemExit.expectSystemExitWithStatus( 0 );
+        Cli.main( new String[] { "--printProjectDeps", "--file", root.getAbsolutePath() } );
+
+        assertThat( systemOutRule.getLog(), equalTo( "INFO org.commonjava.maven.ext.cli.Cli - Found 82 dependencies\n" + "ch.qos.logback:logback-classic:1.2.3                                            jar                 compile             \n"
```

Comment:
There is a newline issue here since `System.out.println()` line separator varies.

You could just use `getLogWithNormalizedLineSeparator()` instead of `getLog()` here.

It would be more accurate to change `\n` to `String.format()` with `%n` because that would verify the "correct" line separator vs. ignoring it.

Reply:
Its partly because I copy and paste the text in intellij and it adds the separators. I can use `getLogWithNormalizedLineSeparator`.